### PR TITLE
Celements 4 Preparation

### PR DESCRIPTION
**Pull request by msladek**
https://synjira.atlassian.net/browse/CELDEV-811

### CODE REVIEW

File: base-pom/pom.xml [hunk 1]
```diff
@@ -205,7 +205,7 @@
       <dependency>
         <groupId>org.apache.xmlrpc</groupId>
         <artifactId>xmlrpc-server</artifactId>
-        <version>3.1</version>
+        <version>3.1.3</version>
```
Comment (fpichler) [inline base-pom/pom.xml:208]: what do we use the apache xmlrpc for? Can we drop this dependency (reason: https://www.cvedetails.com/cve/CVE-2016-5002/)

File: base-pom/pom.xml [hunk 2]
```diff
@@ -256,40 +256,39 @@
...
         <groupId>commons-beanutils</groupId>
         <artifactId>commons-beanutils</artifactId>
-        <version>1.8.3</version>
+        <version>1.9.3</version>
       </dependency>
       <dependency>
         <groupId>commons-fileupload</groupId>
         <artifactId>commons-fileupload</artifactId>
-        <version>1.2.1</version>
+        <version>1.3.3</version>
```
Comment (fpichler) [inline base-pom/pom.xml:275]: please update to latest released version 1.4 (see http://commons.apache.org/proper/commons-fileupload/ )

File: base-pom/pom.xml [hunk 3]
```diff
@@ -299,13 +298,13 @@
...
         <artifactId>xwiki-plugin-scheduler</artifactId>
-        <version>1.24</version>
+        <version>1.25</version>
       </dependency>
 
       <dependency>
         <groupId>javax.servlet</groupId>
         <artifactId>servlet-api</artifactId>
-        <version>2.4</version>
+        <version>2.5</version>
```
Comment (fpichler) [inline base-pom/pom.xml:309]: I suppose we need to change to Tomcat 9.0.x and Servlet-Api Spec 4.0 (see: http://tomcat.apache.org/whichversion.html )

File: base-pom/pom.xml [hunk 4]
```diff
@@ -362,48 +360,55 @@
       <dependency>
         <groupId>org.easymock</groupId>
         <artifactId>easymock</artifactId>
-        <version>3.1</version>
+        <version>3.6</version>
```
Comment (fpichler) [inline base-pom/pom.xml:366]: Suggest trying to upgrade to easymock 4.0.2 (runs on Java 8+ see https://github.com/easymock/easymock/releases/tag/easymock-4.0 )

File: base-pom/pom.xml [hunk 5]
```diff
@@ -362,48 +360,55 @@
...
       </dependency>
       <dependency>
         <groupId>org.easymock</groupId>
         <artifactId>easymockclassextension</artifactId>
-        <version>3.1</version>
+        <version>3.2</version>
```
Comment (fpichler) [inline base-pom/pom.xml]: we can probably drop the classextension because it is integrated in the main jar.

File: base-pom/pom.xml [hunk 6]
```diff
@@ -362,48 +360,55 @@
...
         <groupId>net.sourceforge.htmlunit</groupId>
         <artifactId>htmlunit</artifactId>
-        <version>2.8</version>
+        <version>2.35.0</version>
       </dependency>
       <dependency>
         <groupId>org.seleniumhq.selenium</groupId>
         <artifactId>selenium-java</artifactId>
-        <version>2.31.0</version>
+        <version>2.53.1</version>
```
Comment (fpichler) [inline base-pom/pom.xml]: suggest upgrading to latest 3.141.59

File: components-pom/pom.xml [hunk 7]
```diff
@@ -267,7 +267,7 @@
     <dependency>
       <groupId>com.google.code.findbugs</groupId>
       <artifactId>jsr305</artifactId>
-      <version>3.0.1</version>
+      <version>3.0.2</version>
```
Comment (fpichler) [inline components-pom/pom.xml:270]: Can we change to SpotBugs as findbugs is not maintained anymore? see https://gleclaire.github.io/findbugs-maven-plugin/
and https://spotbugs.readthedocs.io/en/latest/maven.html

**Review by fpichler — APPROVED** [review]

**Review by fpichler — CHANGES_REQUESTED** [review]
please see comments about latest versions

**Review by fpichler — CHANGES_REQUESTED** [review]
please see comments about latest versions

File: base-pom/pom.xml (hunk 2 above)
Comment (fpichler) [reply base-pom/pom.xml:275]: we keep 1.3.3 for the moment. See https://synjira.atlassian.net/browse/CELDEV-812

File: base-pom/pom.xml (hunk 1 above)
Comment (msladek) [reply base-pom/pom.xml:208]: XWikiContext still holds dependency to xmlrpc for some reason

File: base-pom/pom.xml (hunk 4 above)
Comment (msladek) [reply base-pom/pom.xml:366]: 4.x removed deprecations we're still using, e.g. `new Capture()` -> `EasyMock.newCapture()`

File: components-pom/pom.xml (hunk 7 above)
Comment (msladek) [reply components-pom/pom.xml:270]: I can't find a spotbugs jsr305 artifact. it's still the most currently used one, see:
https://mvnrepository.com/search?q=jsr305
does it need to be maintained? it simply contains some annotations we use, e.g. `@NotNull`

File: base-pom/pom.xml (hunk 3 above)
Comment (msladek) [reply base-pom/pom.xml:309]: older tomcat version also support java8, note the "x and later" remark. I wasn't comfortable with updating a major version here.

File: base-pom/pom.xml (hunk 4 above)
Comment (fpichler) [reply base-pom/pom.xml:366]: please add comment for  `TODO: https://synjira.atlassian.net/browse/CELDEV-813`

File: components-pom/pom.xml (hunk 7 above)
Comment (fpichler) [reply components-pom/pom.xml:270]: Added issue https://synjira.atlassian.net/browse/CELDEV-814

File: base-pom/pom.xml (hunk 3 above)
Comment (fpichler) [reply base-pom/pom.xml:309]: added https://synjira.atlassian.net/browse/CELDEV-815

**Review by fpichler — CHANGES_REQUESTED** [review]

File: base-pom/pom.xml (hunk 4 above)
Comment (msladek) [reply base-pom/pom.xml:366]: done

**Review by fpichler — APPROVED** [review]

**Pull request merged**